seqr/views/utils/dataset_utils.py

```diff
@@ -401,30 +395,41 @@ def _load_rna_seq(model_cls, file_path, user, mapping_file, ignore_extra_samples
 
             samples_by_id[(sample_id, project)][gene_id] = row_dict
 
+    conflict_messages = []
+    for project, sample in samples_with_conflict_tissues.items():
+        messages = []
+        for sample_id, tissue_types in sample.items():
+            sample_id_to_tissue_type.pop((sample_id, project))
+            samples_by_id.pop((sample_id, project))
+            messages.append(f'{sample_id} ({", ".join(sorted([REVERSE_TISSUE_TYPE[tissue_type] for tissue_type in tissue_types]))})')
+        conflict_messages.append(f'{", ".join(messages)} in the project {project}')
+    num_samples = sum([len(v) for v in samples_with_conflict_tissues.values()])
+    warnings = [f'Skipped loading for the following {num_samples} tissue type unmatched sample(s) in '
+                f'{len(samples_with_conflict_tissues)} project(s): {", ".join(conflict_messages)}'] if num_samples else []
+
     message = f'Parsed {len(samples_by_id)} RNA-seq samples'
     info = [message]
     logger.info(message, user)
 
     data_source = file_path.split('/')[-1].split('_-_')[-1]
-    samples, remaining_sample_ids = _match_and_update_rna_samples(
+    existing_samples, samples, remaining_sample_ids = _match_and_update_rna_samples(
         projects=get_internal_projects(),
         user=user,
         sample_project_tuples=samples_by_id.keys(),
         data_source=data_source,
         sample_id_to_individual_id_mapping=sample_id_to_individual_id_mapping,
-        raise_unmatched_error_template=None if ignore_extra_samples else 'Unable to find matches for the following samples: {sample_ids}'
+        raise_unmatched_error_template=None if ignore_extra_samples else 'Unable to find matches for the following samples: {sample_ids}',
+        sample_id_to_tissue_type=sample_id_to_tissue_type,
     )
 
-    warnings = []
-    if validate_samples:
-        samples = validate_samples(samples, sample_id_to_tissue_type, warnings)
-
     # Delete old data
     to_delete = model_cls.objects.filter(sample__in=samples).exclude(sample__data_source=data_source)
     prev_loaded_individual_ids = set(to_delete.values_list('sample__individual_id', flat=True))
     if to_delete:
         model_cls.bulk_delete(user, to_delete)
 
+    Sample.bulk_update(user, {'data_source': data_source}, id__in={s.id for s in existing_samples})
```

Comment:
I thought we discussed filtering this so it would not update samples that already had a matched data source?

Reply:
I updated the `bulk_update` function in the `models.py`. See this [comment](https://github.com/broadinstitute/seqr/pull/3243/files#r1154823791).